sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/conditionals.scala

```diff
@@ -312,3 +313,103 @@ case class CaseKeyWhen(key: Expression, branches: Seq[Expression]) extends CaseW
     }.mkString
   }
 }
+
+case class Least(children: Seq[Expression]) extends Expression {
```

Comment:
can you add scaladoc explaining the semantics of this expression? also do it for greatest.